### DISCUSSION TITLE
How to use the -flatten and -page in imageMagick to overlap one image on another image with specific offset value

### DISCUSSION THREAD

**Discussion by koturwaraditi** [ImageMagick/ImageMagick#3068] [General]
MY program use imageMagick to resize actual image by keeping the aspect ratio and adding the smaller marker image on the actual image

First I am keep the height 150 and width would be float to keep the aspect ratio of the original image and later I am finding the relative point where I need to keep the marker pointer Below is the code what I have use

def originalImageAspectRatio = Integer.parseInt(params.imageWidth)/Integer.parseInt(params.imageHeight)

def width = 150 * originalImageAspectRatio

        IMOperation op = new IMOperation();
        op.addImage("-")
        op.quality(50)
        op.strip();
        op.autoOrient();
        op.addRawArgs("-resize",width+"x"+150+"^")
        op.page(0,0,(int)relativePoint.x,(int)relativePoint.y)
        op.addImage(markerIcon)
        op.flatten()
        op.addImage("jpeg:-")      
         
Here I want to create a thumbnail image and  marker pointer on the thumbnail image and it should be same as the original image contain so for that I am finding the relative point of x and y

But the image is resizing properly and also keeping the aspect ration but the marker pointer on the thumbnail image is not coming at the same coordinate point where I have added in the large image

**Comment by snibgo** [comment]
I'm not sure what your program is trying to do, or what is going wrong. I don't even know what language you are using.

But you are applying "page" to the first image. Perhaps the result you want will come from applying it to the second image, like this command:
```
magick - -quality 50 -strip -auto-orient -resize $widthx150^ ( marker.png -page 0x0+40+50 ) -flatten jpeg:
```

**Comment by snibgo** [comment]
Sorry, I don't use Java and can't advise.
